File: 42584.cpp
// 주식가격

#include <string>
#include <vector>

using namespace std;

vector<int> solution(vector<int> prices) {
    vector<int> answer;

    for (int i = 0; i < prices.size(); i++) {
        int tmp = prices[i];
        int idx = i;

        while (idx < prices.size()) {
            if (idx == (prices.size() - 1)) {
                break;
            }
            if (prices[idx] < tmp) {
                break;
            } else { idx++; }
        }

        answer.push_back(idx - i);
    }
    return answer;
}

File: 42584.py
# 주식가격

def solution(prices):
    answer = []
    m = 1
    for idx, p in enumerate(prices):
        if p == 1:
            answer.append(len(prices) - idx - 1)
        elif idx == (len(prices) - 1):
            answer.append(0)
        else:
            tmp = idx + 1
            for x in range(tmp, len(prices)):
                if prices[x] < p:
                    answer.append(x - idx)
                    break
            else:
                answer.append(len(prices)- idx -1)
    return answer